### ISSUE TITLE
can you please provide inference file for embeddings ?

### ISSUE THREAD

**Issue by Neel125** [AmirSh15/FECNet#1]
Using this method i want to get embeddings from the model for query image. So, can you please give me some instruction for how to get embeddings from model.

**Comment by AmirSh15** [comment]
It's nothing unusual using this network. Like other PyTorch networks, you should first load weights from pretrained model then feed your image data and get the embedding. Note that your image should be clear as the training data, I mean the image data are close-up. Otherwise, you must use some face detector network for getting the close-up face images such as dlib library. The below code may help you:

model = MyModel()
model.eval()
image_dir = "an string dir of your images"
with torch.no_grad():
	for i, image_path in enumerate(tqdm(image_dir.glob("*.jpg"))):
			img = cv2.imread(image_path)
            img = trans(Image.fromarray(img))
            img = torch.Tensor(img).view(1,3,224,224)
            Embedding = model(img.cuda())

**Comment by Neel125** [comment]
Thank You @AmirSh15 for quick reply.

**Comment by AmirSh15** [comment]
You are welcome. Yes, I tried. I'll refer the paper after getting accepted.

**Comment by Neel125** [comment]
So, What about the result? I mean how good are these embeddings for emotion classification and which data used for emotion classification?

**Comment by AmirSh15** [comment]
I used the embeddings for RML, eNTERFACE, and RAVDESS databases and got recognition performance of 71,79,71% respectively.

**Comment by Neel125** [comment]
Using these embeddings and data which model you trained like SVM, Neural Net, KNN, etc? I am also working on Emotion classification Projects and I trained cnn but it did not give the accuracy.

**Comment by AmirSh15** [comment]
Sorry, I can't provide any detail, but I trained on a CNN based model.

**Comment by Jun12Shao** [comment]
Hello, do you train your model(validation accuracy 64.3) on the whole  Triplet dataset or only the "strong" set?

**Comment by AmirSh15** [comment]
Hi, on the whole test set.

**Comment by ajayrfhp** [comment]
Hi, not sure if I follow correctly. Do you provide the weights for trained model in the repository ? I couldnt find any links.

**Comment by AmirSh15** [comment]
Hi,
If you run the code, it will download and use the pretrained weights automatically.
